src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -57,6 +57,15 @@ internal TagHelperExecutionContext(string tagName)
         }
 
         /// <summary>
+        /// Indicates whether or not the tag is self closing.
+        /// </summary>
+        public bool SelfClosing
```

Comment:
@dougbu 

1. The `TagHelperExecutionContext` represents the current state of the HTML tag it represents (and shouldn't be changed) therefore `SelfClosing` makes sense here. As for discussing another name, we spit balled the idea a bit but decided to leave it as is from my recollection.
1. :+1: 
1. I don't know if I'd go as far as mentioning the `TagHelperOutput` here because the `TagHelperExecutionContext` has no direct correlation to the `TagHelperOutput` object. The `TagHelperRunner` is the mechanism responsible for joining the two. As for adding a remark stating that this is the "original" value, it should never change since the `TagHelperExecutionContext` is by definition the metadata object for the HTML element, addressing 4 should solve this confusion.
1. I agree, it shouldn't be settable. Does kind of suck though since this means it's an extra property we need to pass into `TagHelperScopeManager.Begin` (that constructs the `TagHelperExecutionContext`).